src/Analyzers/CSharp/CodeFixes/UseIndexOrRangeOperator/CSharpUseRangeOperatorCodeFixProvider.cs

```diff
@@ -119,6 +119,12 @@ private static ExpressionSyntax FixOne(Result result, SyntaxGenerator generator)
                     return ElementBindingExpression(argumentList);
                 }
 
+                if (invocation.Expression is IdentifierNameSyntax)
+                {
+                    // Substring(...) -> this[...]
+                    return ElementAccessExpression(ThisExpression(), argumentList);
```

Comment:
i'd prefer we also validate that we're inside the string class.  Or, if htis is handling more than just being inside System.String and is handling things like .Slice inside other types, then we can have a simpler test that just shows that.

Reply:
I'd prefer the latter, adding a test for slicing on self.